OrbitQt/ProfilingTargetDialog.cpp

```diff
@@ -0,0 +1,493 @@
+ProfilingTargetDialog::ProfilingTargetDialog(
+    SshConnectionArtifacts* ssh_connection_artifacts, MainThreadExecutor* main_thread_executor,
+    std::optional<ConnectionConfiguration> connection_configuration, QWidget* parent)
+    : QDialog(parent),
+      ui_(std::make_unique<Ui::ProfilingTargetDialog>()),
+      current_target_(TargetEnum::kFile),
+      main_thread_executor_(main_thread_executor),
+      local_grpc_port_(ssh_connection_artifacts->GetGrpcPort().grpc_port),
+      s_stadia_(&state_machine_),
+      s_s_history_(&s_stadia_),
+      s_s_connecting_(&s_stadia_),
+      s_s_connected_(&s_stadia_),
+      s_s_processes_loading_(&s_s_connected_),
+      s_s_process_selected_(&s_s_connected_),
+      s_s_no_process_selected_(&s_s_connected_),
+      s_file_(&state_machine_),
+      s_f_history_(&s_file_),
+      s_f_file_selected_(&s_file_),
+      s_f_no_file_selected_(&s_file_),
+      s_local_(&state_machine_),
+      s_l_history_(&s_local_),
+      s_l_connecting_(&s_local_),
+      s_l_connected_(&s_local_),
+      s_l_processes_loading_(&s_l_connected_),
+      s_l_process_selected_(&s_l_connected_),
+      s_l_no_process_selected_(&s_l_connected_) {
+  CHECK(ssh_connection_artifacts != nullptr);
+  CHECK(main_thread_executor_ != nullptr);
+
+  setWindowFlags(Qt::Window);
+
+  ui_->setupUi(this);
+
+  SetupStateMachine();
+
+  process_proxy_model_.setSourceModel(&process_model_);
+  process_proxy_model_.setSortRole(Qt::EditRole);
+  process_proxy_model_.setFilterCaseSensitivity(Qt::CaseInsensitive);
+  ui_->processesTableView->setModel(&process_proxy_model_);
+  ui_->processesTableView->setSortingEnabled(true);
+  ui_->processesTableView->sortByColumn(static_cast<int>(ProcessItemModel::Column::kCpu),
+                                        Qt::DescendingOrder);
+
+  ui_->processesTableView->horizontalHeader()->resizeSection(
+      static_cast<int>(ProcessItemModel::Column::kPid), 60);
+  ui_->processesTableView->horizontalHeader()->resizeSection(
+      static_cast<int>(ProcessItemModel::Column::kCpu), 60);
+  ui_->processesTableView->horizontalHeader()->setSectionResizeMode(
+      static_cast<int>(ProcessItemModel::Column::kName), QHeaderView::Stretch);
+  ui_->processesTableView->verticalHeader()->setDefaultSectionSize(kProcessesRowHeight);
+  ui_->processesTableView->verticalHeader()->setVisible(false);
+
+  if (absl::GetFlag(FLAGS_local)) {
+    ui_->localFrame->setVisible(true);
+  }
+
+  QObject::connect(ui_->loadFromFileButton, &QPushButton::clicked, this,
+                   &ProfilingTargetDialog::SelectFile);
+  QObject::connect(ui_->loadCaptureRadioButton, &QRadioButton::clicked, this, [this](bool checked) {
+    if (!checked) ui_->loadCaptureRadioButton->setChecked(true);
+  });
+  QObject::connect(ui_->localProfilingRadioButton, &QRadioButton::clicked, this,
+                   [this](bool checked) {
+                     if (!checked) ui_->localProfilingRadioButton->setChecked(true);
+                   });
```

Comment:
This is to mimic the behaviour of a exclusive (radio) button group. When a button is was unchecked and the user clicks on it, it will change to checked and the state machine will transition into the correct state. If a button is checked and the user clicks on it again, this code here, will prevent the button from actually getting unchecked. 

Maybe there is a way to create a QButtonGroup, even though one of the three radio buttons is in another Widget (ConnectToStadiaWidget). I guess an alternative could also be to not have the "Connect to a stadia instance" radio button in ConnectToStadiaWidget, but be part of the ProfilingTargetWidget. And then a QButtonGroup is definitely possible and would remove the need for these connections. 
wdyt?

Reply:
Thanks a lot for the explanation. Makes sense now.

So regarding the best solution here:
1. I can live with that design, but it needs an explanation in terms of a comment.
2. QButtonGroup should work as well. It only needs pointers to all the buttons, which means we need to expose an implementation detail from the ConnectToStadiaWidget which is not super nice.
3. Moving the radio button out of the ConnectToStadiaWidget would be my favorite solution - if that's possible. That makes it design-wise better, and then a QButtonGroup could be used easily.

Reply:
I have not done anything about this yet. Depending on the time I have, I might implement the 3rd option, or just comment on it (1.)
I will not resolve this conversation so I dont forget it.

Reply:
I did only comment on it